convert2rhel/utils.py

```diff
@@ -620,4 +627,59 @@ def set_locale():
     os.environ.update({"LC_ALL": "C", "LANG": "C"})
 
 
+def options_from_config_files(cfg_path=None):
+    """Get the options from config file.
+
+    The order is:
+    1. if filename is set then load from it
+    2. ~/.convert2rhel.ini
+    3. /etc/convert2rhel.ini
+
+    Return dict of supported options with their values.
+    """
+    headers = ["subscription_manager"]  # supported sections in config file
+    config_file = configparser.ConfigParser()
+    paths = CONFIG_PATHS
+    # Create dict with all supproted options, all of them set to None
+    # needed for avoiding problems with files priority
+    # The name of supported option MUST correspond with the name in ToolOpts()
+    # Otherwise it won't be used
+    supported_opts = {"password": None, "activation_key": None}
+
+    if cfg_path:
+        if not os.path.exists(cfg_path):
+            raise OSError(2, "No such file or directory: '" + cfg_path + "'")
+        paths.insert(0, cfg_path)  # highest priority
+
+    for path in paths:
+        if os.path.exists(path):
+            if not oct(os.stat(path).st_mode)[-4:].endswith("00"):
+                loggerinst.critical("The %s file must only be accessible by the owner (0600)" % path)
+            config_file.read(path)
+
+            for header in config_file.sections():
+                if header in headers:
+                    for option in config_file.options(header):
+                        if option.lower() in supported_opts:
+                            # Solving priority
+                            if supported_opts[option.lower()] is None:
+                                supported_opts[option] = config_file.get(header, option)
+                                loggerinst.debug("Found %s in %s" % (option, path))
+                        else:
+                            loggerinst.warning("Unsupported option %s in %s" % (option, path))
+                elif header not in headers and header != "DEFAULT":
+                    loggerinst.warning("Unsupported header %s in %s." % (header, path))
+
+    return supported_opts
+
+
+def set_opts(opts, supported_opts):
+    """Set ToolOpts data using dict with values from config file."""
+    for key, value in supported_opts.items():
+        if value is not None and hasattr(opts, key):
+            setattr(opts, key, value)
+
+    return opts
```

Comment:
**Todo Now**: Seeing this broken out into its own function, I think that it would be good as a method of the `ToolOpts` class because:

* The first argument is an instance of `ToolOpts`.
* It mutates the attributes stored in the `ToolOpts`.

**Possible Future Refactor:** One thing that will be a little odd is that setting the cfg data on the `ToolOpts` will then be a method of the `ToolOpts` class but setting the command line arguments will be handled by the `CLI` class. I don't think that's a problem with your code, though.  Instead, I think it means we should look at changing the interface between the `CLI` class and the `ToolOpts` in the future.  

Maybe that eventual refactor should look more like:
``` python
    class ToolOpts
        def__init__(self,
                    args: List = sys.argv[1:],
                    cfg_files: List[Path] = CONFIG_PATHS):
        """
        Instantiate a toolopts, setting the values from configuration files and the programs command
        line arguments.

        Path in cfg_files which is not in CONFIG_PATHS must exist.
        """
```

`ToolOpts` could then call out to the `CLI` class to parse the arguments or the logic from `CLI` could
move into helper methods on `ToolOpts`.  I'm not sure which would be better without looking more closely.